aws-api/src/main/java/com/amplifyframework/api/aws/GraphQLResponseDeserializer.java

```diff
@@ -89,6 +127,42 @@ private JsonElement skipQueryLevel(JsonElement jsonData) throws JsonParseExcepti
         return data.get(data.keySet().iterator().next());
     }
 
+    private String getModelName(String queryName) {
+        // TODO the right parsing (This assumes a sync query with plural name by appending s
+        final int syncQueryStartIndex = 4; // 4 chars in sync
+        final int createQueryStartIndex = 6; // 6 chars in create
+        final int onCreateQueryStartIndex = 8; // 6 chars in create
+        if (queryName.substring(0, syncQueryStartIndex).equals("sync")) {
+            return queryName.substring(syncQueryStartIndex, queryName.length() - 1);
+        } else if (queryName.substring(0, createQueryStartIndex).equals("create")) {
+            return queryName.substring(createQueryStartIndex);
+        } else if (queryName.substring(0, onCreateQueryStartIndex).equals("onCreate")) {
+            return queryName.substring(onCreateQueryStartIndex);
+        }
+        return null;
+    }
```

Comment:
Hmm.. it's a good idea! Please forgive if a noob question, I'm unable to visualize how it will work for Subscription events. For instance assume there are three model types in client code and they started subscription for all three. When an event is received, how does this figure out which model/request it belonged to?

Reply:
It should work in a similar way to how the queries and mutations work.  The `SubscriptionOperation` object remains in memory for the lifecycle of the subscription, and holds a reference to the original `GraphQLRequest` which started the subscription.   

There is one other small thing you'll need to do, specifically related to subscriptions though - We currently pass `null` for the `GraphQLRequest` parameter to the response factory for parsing the response from a subscription here: https://github.com/aws-amplify/amplify-android/blob/71aa76d145812170d2f9d8e3110ef907427e08ef/aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java#L370.  You'll need to update that to pass in the actual `GraphQLRequest` itself, which shouldn't be too hard.  As mentioned, it is available on the `SubscriptionOperation` object.

Reply:
Ahh, that makes sense. Made my life so much easy!